Realm/Realm/Transaction.cs

```diff
@@ -74,10 +75,23 @@ public void Commit()
 
         /// <summary>
         /// Use to save the changes to the realm. If <see cref="Transaction"/> is declared in a <c>using</c> block,
-        /// it must be used before the end of that block.
+        /// it must be used before the end of that block. It terminates when the changes are effectively written to disk.
         /// </summary>
-        /// <returns>An awaitable <see cref="Task"/>.</returns>
-        public async Task CommitAsync()
+        /// <remarks>
+        /// Cancelling the returned <see cref="Task"/> simply makes the <see cref="Task"/>'s execution continue;
+        /// it does not cancel the commit action.
```

Comment:
This language construct is a bit awkward. I'd suggest saying something like "Cancelling the returned Task will not prevent the write to disk but will immediately resolve the task with a `TaskCanceledException`".